apps/management-app/src/components/Authenticated/Officer/PanelEditing/index.tsx

```diff
@@ -0,0 +1,417 @@
+        width: panelData?.data?.width,
+        height: panelData?.data?.height,
+        createContractDate: panelData?.data?.createContractDate,
+        expiredContractDate: panelData?.data?.expiredContractDate,
+        companyEmail: panelData?.data?.companyEmail,
+        companyNumber: panelData?.data?.companyNumber,
+        reason: '',
+      });
+    }
+  }, [panelData, panelTypeData, reset, userId]);
+
+  useEffect(() => {
+    if (panel?.imageUrls) {
+      fetch(panel?.imageUrls[0])
+        .then((res) => res.blob())
+        .then((blob) => {
+          const file = new File([blob], panel?.imageUrls[0]);
+          setValue('images', [file]);
+        });
+    }
+  }, [panel?.imageUrls, setValue]);
+
+  const { errors: formError } = formState;
+  const formValue = watch();
+
+  const [submitting, setSubmitting] = useState(false);
+
+  const handleAddImage = useCallback(
+    (file: File) => setValue('images', [...formValue.images, file]),
+    [formValue.images, setValue],
+  );
+
+  const handleUpdateImage = useCallback(
+    (file: File | null) => {
+      if (!file) {
+        return;
+      }
+
+      dispatch(
+        showModal(ModalKey.CROP_IMAGE, {
+          image: file,
+          aspect: 16 / 9,
+          onSubmit: handleAddImage,
+        }),
+      );
+    },
+    [dispatch, handleAddImage],
+  );
+
+  const handleDeleteImage = useCallback(
+    (file: File) => {
+      setValue(
+        'images',
+        formValue?.images.filter((image) => image !== file),
+      );
+    },
+    [formValue.images, setValue],
+  );
+
+  const renderUpdateImageContainer = ({
+    open,
+    disabled,
+  }: {
+    open: () => void;
+    disabled: boolean;
+  }) => <UploadImageCard open={open} disabled={disabled} />;
+
+  const [updatePanel] = useCreateUpdatePanelRequestMutation();
+
+  const onSubmit = async (data: UpdatePanelDto) => {
+    try {
+      setSubmitting(true);
+
+      const formData = new FormData();
+      formData.append('belongPanelId', data.belongPanelId.toString());
+      formData.append('locationId', data.locationId.toString());
+      formData.append('userId', data.userId.toString());
+      formData.append('typeId', data.typeId.toString());
+      data.images.forEach((image) => formData.append('images', image));
+      formData.append('reason', data.reason);
+      formData.append('width', data.width.toString());
+      formData.append('height', data.height.toString());
+      data.createContractDate = new Date(data.createContractDate).toISOString();
+      data.expiredContractDate = new Date(
+        data.expiredContractDate,
+      ).toISOString();
+      formData.append('createContractDate', data.createContractDate);
+      formData.append('expiredContractDate', data.expiredContractDate);
+      formData.append('companyEmail', data.companyEmail);
+      formData.append('companyNumber', data.companyNumber);
+
+      await updatePanel(formData).unwrap();
+
+      setSubmitting(false);
+
+      navigate(-1);
```

Comment:
refactor this and add showSuccess toaster on updated API

Reply:
Can we use **getOnMutationFunction** in _helper.ts_?
![image](https://github.com/starhuy1903/ads-management/assets/90955782/d449d2d3-5e1e-4532-a169-762bf6e46a1d)
![image](https://github.com/starhuy1903/ads-management/assets/90955782/9d5cdd90-8c68-43bd-aac8-ecdd9abba664)

Reply:
should `showSuccess` in component and we will refactor all code later